src/components/generator/EditDialog.js

```diff
@@ -0,0 +1,130 @@
+import FetchHandler from "../common/FetchHandler";
+import Settings from "../common/settings";
+import Notifications from "../common/Notifications";
+import Editor from "../editor/Editor";
+import { Button, Dialog, DialogActions, DialogContent, DialogTitle, IconButton, styled } from '@mui/material';
+import CloseIcon from '@mui/icons-material/Close';
+import { Link, useNavigate, useParams } from "react-router-dom";
+
+const EditDialogRoot = styled('div')({
+  width: '100%',
+  display: 'flex',
+  flexDirection: 'column',
+  alignItems: 'center',
+
+  '> *': {
+    marginBottom: '16px',
+  }
+});
+
+const StyledButton = styled(Button)({
+  cursor: 'pointer',
+  width: '128px',
+
+  ':hover': {
+    backgroundColor: '#BFBFBF',
+  },
+});
+
+const StyledIconButton = styled(IconButton)({
+    position: 'absolute',
+    top: 10,
+    right: 0,
+  });
+
+const synchronizeProject = (weldingPoints) => {
+  return FetchHandler.simple(fetch(
+      `${Settings.weldingPointsPath}`,
+      {method: "PUT", body: JSON.stringify(weldingPoints), headers: {"Content-Type": "application/json"}}
+  ));
+};
+
+const EditDialog = () => {
+    const [projectRetrievalState, setProjectRetrievalState] = useState("loading");
+    const [selectedProject, setSelectedProject] = useState(undefined);
+    const [weldingPoints, setWeldingPoints] = useState([]);
+    const [robots, setRobots] = useState([]);
+    const [open, setOpen] = useState(true);
+
+    let navigate = useNavigate();
+    let { id } = useParams();
+
+    useEffect(() => {
+      FetchHandler.readingJson(fetch(Settings.projectPath + id, {method: "GET"}))
+          .then((project) => {
+              setSelectedProject(project);
+              setProjectRetrievalState("success");
+          })
+          .catch((err) => {
+              Notifications.notify(`Failed to retrieve projects\n${err}`, "error");
+              setProjectRetrievalState("failed");
+          });
+    }, [id]);
+
+      useEffect(() => {
+        FetchHandler.readingJson(fetch(Settings.robotPath, {method: "GET"}))
+            .then((robots) => {
+                setRobots(robots);
+                setProjectRetrievalState("success");
+            })
+            .catch((err) => {
+                Notifications.notify(`Failed to retrieve robots\n${err}`, "error");
+                setProjectRetrievalState("failed");
+            });
+      }, []);
+
+    const onUpdate = () => synchronizeProject(weldingPoints)
+        .then(() => Notifications.notify("Synchronized project", "success"))
+        .catch((err) => Notifications.notify(`Failed to synchronize project\n${err}`));
+
+    const closeEdit = () => {
+      setOpen(false);
+      navigate("/generate");
+    };
+
+    return (
+      <EditDialogRoot>
+        {(projectRetrievalState === "success" && selectedProject !== undefined) ? (
+          <Dialog
+            open={open}
+            onClose={closeEdit}
+            maxWidth={'lg'}
```

Comment:
```suggestion
            maxWidth="lg"
```
Static property suffices

Reply:
implemented in branch: "generatorpage-add-delete"